volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -33,8 +37,38 @@ def get_requirements(cls):
             ),
         ]
 
-    def _generator(self):
-        layer = self.context.layers[self.config["primary"]]
+    @staticmethod
+    def enumerate_mft_records(
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        primary_layer_name: str,
+        attr_callback: Callable[
+            [
+                Dict[int, Tuple[str, int, int]],
+                interfaces.objects.ObjectInterface,
+                interfaces.objects.ObjectInterface,
+                str,
+            ],
+            Generator,
+        ],
+    ) -> interfaces.objects.ObjectInterface:
+        try:
+            primary = context.layers[primary_layer_name]
+        except KeyError:
+            vollog.error(
+                "Unable to obtain primary layer for scanning. Please file a bug on GitHub about this issue."
+            )
+            return
+
+        try:
+            phys_layer = primary.config["memory_layer"]
```

Comment:
This is fine, but as a point for the future, this may not work on arm layers, etc...